data_structures/skip_list.cpp

```diff
@@ -0,0 +1,197 @@
+/**
+ * A skip list is a data structure that is used for storing a sorted list of items with a
+ *  help of hierarchy of linked lists that connect increasingly sparse subsequences of the items
+ * 
+ *  References used: GeeksForGeeks skip list code, https://iq.opengenus.org/skip-list/ Pseudo Code.
+*/
+
+#include <iostream>
+#include <vector>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <time.h>   
+#include <cstdlib>
+
+// using namespace std; 
+using std::vector;
+using std::endl;
+
+#define MAXLVL 2
```

Comment:
use descriptive macro names as in [this file](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/numerical_methods/newton_raphson_method.cpp) otherwise, it is difficult for a reader to learn from the code.

Reply:
ok, thank you